src/irr/asset/CGLIWriter.cpp

```diff
@@ -0,0 +1,327 @@
+
+			const auto& imageViewInfo = imageView->getCreationParameters();
+			const auto& imageInfo = imageViewInfo.image->getCreationParameters();
+			const auto& image = imageViewInfo.image;
+
+			if (image->getRegions().length() == 0)
+			{
+				os::Printer::log("WRITING GLI: there is a lack of regions!", file->getFileName().c_str(), ELL_INFORMATION);
+				return false;
+			}
+			if (isBlockCompressionFormat(imageInfo.format))
+			{
+				os::Printer::log("WRITING GLI: the writer doesn't support any compression format!", file->getFileName().c_str(), ELL_INFORMATION);
+				return false;
+			}
+				
+
+			const bool facesFlag = doesItHaveFaces(imageViewInfo.viewType);
+			const bool layersFlag = doesItHaveLayers(imageViewInfo.viewType);
+
+			const bool floatingPointFlag = isFloatingPointFormat(imageInfo.format);
+			const bool integerFlag = isIntegerFormat(imageInfo.format);
+			const bool signedTypeFlag = isSignedFormat(imageInfo.format);
+		
+			const auto texelBlockByteSize = asset::getTexelOrBlockBytesize(imageInfo.format);
+			const auto channelsAmount = getFormatChannelCount(imageInfo.format);
+			const auto singleChannelByteSize = texelBlockByteSize / channelsAmount;
+			const auto data = reinterpret_cast<const uint8_t*>(image->getBuffer()->getPointer());
+
+			auto getTarget = [&]()
+			{
+				switch (imageViewInfo.viewType)
+				{
+					case ICPUImageView::ET_1D: return gli::TARGET_1D;
+					case ICPUImageView::ET_1D_ARRAY: return gli::TARGET_1D_ARRAY;
+					case ICPUImageView::ET_2D: return gli::TARGET_2D;
+					case ICPUImageView::ET_2D_ARRAY: return gli::TARGET_2D_ARRAY;
+					case ICPUImageView::ET_3D: return gli::TARGET_3D;
+					case ICPUImageView::ET_CUBE_MAP: return gli::TARGET_CUBE;
+					case ICPUImageView::ET_CUBE_MAP_ARRAY: return gli::TARGET_CUBE_ARRAY;
+					default: assert(0);
+				}
+			};
+
+			auto getFacesAndLayersAmount = [&]()
+			{
+				size_t layers, faces;
+				const auto arrayLayers = imageInfo.arrayLayers;
+
+				if (layersFlag)
+					layers = ((arrayLayers - 1) / 6) + 1;
+				else
+					layers = 1;
+				if (facesFlag)
+					faces = ((arrayLayers - 1) % 6) + 1;
+				else
+					faces = 1;
+
+				return std::make_pair(layers, faces);
+			};
+
+			auto gliFormatAndSwizzles = getTranslatedIRRFormat(imageViewInfo);
+			gli::target gliTarget = getTarget();
+			gli::extent3d gliExtent3d = {imageInfo.extent.width, imageInfo.extent.height, imageInfo.extent.depth};
+			size_t gliLevels = imageInfo.mipLevels;
+			std::pair<size_t, size_t> layersAndFacesAmount = getFacesAndLayersAmount();
+
+			gli::texture texture(gliTarget, gliFormatAndSwizzles.first, gliExtent3d, layersAndFacesAmount.first, layersAndFacesAmount.second, gliLevels, gli::texture::swizzles_type{ gliFormatAndSwizzles.second[0], gliFormatAndSwizzles.second[1], gliFormatAndSwizzles.second[2], gliFormatAndSwizzles.second[3] });
+
+			auto getFullSizeOfLayer = [&](const uint16_t mipLevel) -> uint64_t
+			{
+				auto region = image->getRegions().begin() + mipLevel;
```

Comment:
that's absolutely not how regions work!!!!!

Reply:
also if you need the dimensions of a mip-level just use https://github.com/buildaworldnet/IrrlichtBAW/blob/shader_pipeline/include/irr/asset/IImage.h#L403

Reply:
ok actually its fine, I see what you're doing

Reply:
its a size of region, not the whole layer.